src/libcollections/vec.rs

```diff
@@ -1380,7 +1380,17 @@ impl<T> FromIterator<T> for Vec<T> {
     fn from_iter<I:Iterator<Item=T>>(iterator: I) -> Vec<T> {
         let (lower, _) = iterator.size_hint();
         let mut vector = Vec::with_capacity(lower);
-        for element in iterator {
+
+        let mut i = iterator.fuse();
+        for element in i.by_ref().take(vector.capacity()) {
+            let len = vector.len();
```

Comment:
Nah re-loading isn't required or necessary, we could keep a separate variable, this was just a pattern used elsewhere in `vec.rs` so I figured I'd do the same.